hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioAsyncServerSocket.java

```diff
@@ -32,29 +33,20 @@
 import java.util.concurrent.TimeUnit;
 import java.util.function.Consumer;
 
-import static com.hazelcast.internal.tpc.util.Util.closeResource;
+import static com.hazelcast.internal.tpc.util.IOUtil.closeResource;
+import static com.hazelcast.internal.tpc.util.ReflectionUtil.findStaticFieldValue;
 import static java.net.StandardSocketOptions.SO_RCVBUF;
 import static java.net.StandardSocketOptions.SO_REUSEADDR;
 import static java.nio.channels.SelectionKey.OP_ACCEPT;
 
 /**
- * Nio version of the {@link AsyncServerSocket}.
+ * Nio implementation of the {@link AsyncServerSocket}.
  */
 public final class NioAsyncServerSocket extends AsyncServerSocket {
     private static final int DEFAULT_LATCH_TIMEOUT_SECONDS = 10;
 
     // This option is available since Java 9, so we need to use reflection.
-    private static final SocketOption SO_REUSEPORT;
-
-    static {
-        SocketOption value = null;
-        try {
-            Field field = StandardSocketOptions.class.getField("SO_REUSEPORT");
-            value = (SocketOption) field.get(null);
-        } catch (Exception ignore) {
-        }
-        SO_REUSEPORT = value;
-    }
+    private static final SocketOption SO_REUSEPORT = findStaticFieldValue(StandardSocketOptions.class, "SO_REUSEPORT");
```

Comment:
Should we also print a log message when this option is missing in Java 8, similar to ones we did for NioAsyncSocket?

Reply:
Good idea

Reply:
Done